intelmq/lib/message.py

```diff
@@ -443,6 +445,25 @@ def to_json(self, hierarchical=False, with_type=False, jsondict_as_string=False)
         json_dict = self.to_dict(hierarchical=hierarchical, with_type=with_type)
         return json.dumps(json_dict, ensure_ascii=False)
 
+    def __eq__(self, other) -> bool:
+        """
+        Necessary as we have an additional member harmonization_config and types.
+        The additional checks are only performed for subclasses of Message.
+
+        Comparison with other types e.g. dicts does not check the harmonization_config.
+        """
+        dict_eq = super(Message, self).__eq__(other)
+        if issubclass(type(other), Message):
+            type_eq = type(self) == type(other)
+            harm_eq = self.harmonization_config == other.harmonization_config if hasattr(other, 'harmonization_config') else False
+            if type_eq and dict_eq and harm_eq:
+                return True
+        elif dict_eq:
+            return True
+
```

Comment:
I think its missing a final `return False`.

Reply:
Implicitly it's None here.
According to the docs, if the return value is not boolean, bool() is called.

Pushed an additional commit

Reply:
Yes, its true but I tested here and it gaves me `None`, not `False`. May be I did something wrong.

```
>>> def return_test(other) -> bool:
...   if other == "a":
...     return True
... 
>>> print( return_test("a") )
True
>>> print( return_test("b") )
None
```